generators/app/templates/Procfile.dev

```diff
@@ -1,2 +1,2 @@
-server: node express.js
-client: rails s -p 3000
+server: cd server && rails s -p 3000
+client: cd client && npm start
```

Comment:
Well, you changed the start script to run webpack-dev-server directly, so this means that it will execute webpack-dev-server in production.